### DISCUSSION TITLE
Does Microsoft want us to use syncfusion?

### DISCUSSION THREAD

**Discussion by HimanshuBansal112** [dotnet/maui#26824] [General]
I expected dotnet Maui to be a full fledged framework, which implements new feature while fixing bugs. But it seems we need to use syncfusion to get unavailable basic features. And after reading that Microsoft has partnered with syncfusion, does it means they support them or want us to use them? If yes, I know this is not a big issue, but if in future, anyone's project become successful or huge, they will need to pay syncfusion. Why make a framework that needs other framework to function?

**Comment by JohnHDev** [comment]
What do you need in Syncfusion that you cannot do yourself in .NET Maui?

**Reply by HimanshuBansal112** [reply]
I needed combo box, auto complete and custom dialog, all of these are not natively present in maui.

**Reply by JohnHDev** [reply]
ComboBox isn't native to iOS on Android, there are established design patterns for that workflow. You could use a custom spinner which is supported.
Autocomplete is p!ss easy to write yourself
Custom dialog, every page we write is custom. You need to be more specific.

> Why make a framework that needs other framework to function?

None of the above are *required* to build solid, working apps in Maui.

**Reply by HimanshuBansal112** [reply]
> ComboBox isn't native to iOS on Android, there are established design patterns for that workflow. You could use a custom spinner which is supported.

If I am understanding correctly, custom spinner will open an option list where picker is present on Windows (and will not be below text box as it should) which might not look prettier. It is valid for Android and ios but not valid for cross platform app.

> Autocomplete is p!ss easy to write yourself

I agree with this but thought to mention it, as it was a basic thing that even html has. Here, we might need to store suggestions on device locally but will be able to achieve same functionality.

> Custom dialog, every page we write is custom. You need to be more specific.

From dialogs, I meant popups where we can select files, or give print command, etc. Someone also opened a feature request for it at https://github.com/dotnet/maui/issues/4795

There might be other things/functionalities/features which I might not have came across when building an app.

I know each thing can be implemented on our own but from what I know, even older GUI packages of .net framework such as windows forms app supported several feature. Why people did not said it can be implemented on our own in older frameworks? And a more common thing is if we were to implement several things on our own, we can end up with some bugs, and might not be faster.

In the end, thanks for your reply.

**Comment by wd-mutasim** [comment]
> unavailable basic features

I've spent a lot of time to reinvent the wheel to create [SelectedValue and SelectedValuePath] for [Picker] control to be able to use the same code in WPF, I am surprised that no one asks about this feature.

**Comment by jingliancui** [comment]
syncfusion Maui community toolkit is under mit license. So it's not a problem for your "pay" issue I think.

**Reply by HimanshuBansal112** [reply]
> syncfusion Maui community toolkit is under mit license. So it's not a problem for your "pay" issue I think.

![image](https://github.com/user-attachments/assets/356487a3-1116-42f2-8e1a-42a4d5564589)
From what I understand, it says a medium-sized companies with more employees, or some successful companies with higher revenue need to pay Syncfusion. Even if we remove successful criteria, most companies still fall into medium-sized category and hence, creating an issue with having to pay for Syncfusion.
